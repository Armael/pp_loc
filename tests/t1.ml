
module P = Pp_loc.Position

let src = {|let rec f x =
  if x < 0 then - (f (-x))
  else x+1
|}

let input = Pp_loc.Input.string src

let loc_rec = P.of_line_col 1 5, P.of_line_col 1 8;;
let loc_if = P.of_line_col 2 3, P.of_line_col 2 5;;

Format.printf "loc is: %a@." (Pp_loc.pp ~max_lines:5 ~input) [loc_rec];;
Format.printf "loc is: %a@." (Pp_loc.pp ~max_lines:5 ~input) [loc_if];;

let loc_rec_if = P.of_line_col 1 5, P.of_line_col 2 5;;
Format.printf "loc is: %a@." (Pp_loc.pp ~max_lines:5 ~input) [loc_rec_if];;
