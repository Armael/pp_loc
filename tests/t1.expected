loc is: 1 | let rec f x =
                ^^^
        
loc is: 2 |   if x < 0 then - (f (-x))
              ^^
        
loc is: 1 | ....rec f x =
        2 |   if......................
        
